coremltools/models/utils.py

```diff
@@ -1056,7 +1056,10 @@ def compile_model(model: _Union['_ct.models.MLModel', str, _Model_pb2.Model]) ->
         return _MLModelProxy.compileModel(model)
 
 
-def make_pipeline(*models):
+def make_pipeline(
+        *models: '_ct.models.MLModel',
+        compute_units: _ct.ComputeUnit =_ComputeUnit.ALL
+    ) -> '_ct.models.MLModel':
```

Comment:
BTW, what about using the compute units with which the input models are loaded with? I guess if they are all loaded with a certain compute unit,  then we should just default to loading the pipelined model with the same compute unit.

I think only if the input models are loaded with different compute units, or if the user also wants to change the compute unit, only then the `compute_units` flag is relevant. 
What I am saying is that we can make this flag as optional (default it to None). If provided then use it to load the newly created model, if not provided then use the compute unit from the input models, if they dont have the same compute unit, then error out and ask for the user to provide a value.

Reply:
Good idea! I've made the change.

Please take another look. The logic is a little hairy. 

To properly test this change, I decided to add a dedicated unit test. I reverted my previous unit test changes. So to review the unit test changes, it's probably easiest to just look at the overall diff rather than the most recent commit diff.